pkg/kube/cluster-init.sh

```diff
@@ -0,0 +1,145 @@
+#!/bin/ash
+#
+# Copyright (c) 2023 Zededa, Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+K3S_VERSION=v1.26.3+k3s1
+KUBEVIRT_VERSION=v0.59.0
+LONGHORN_VERSION=v1.4.2
+CDI_VERSION=v1.56.0
+
+INSTALL_LOG=/var/log/install.log
+date >> $INSTALL_LOG
+
+logmsg() {
+   local MSG
+   local TIME
+   MSG="$*"
+   TIME=$(date +"%F %T")
+   echo "$TIME : $MSG"  >> $INSTALL_LOG
+}
+
+setup_cgroup () {
+   echo "cgroup /sys/fs/cgroup cgroup defaults 0 0" >> /etc/fstab
+}
+
+
+                       
+check_network_connection () {
+ while true; do
+
+   IP="8.8.8.8"
+   if ping -c 5  $IP > /dev/null 2>&1; then
+        logmsg "Network available, ping to $IP was successful."
+        break;
+   else
+        logmsg "Ping to $IP failed."
+   fi
+
+   sleep 5
+ done
+}                                                               
+
+wait_for_default_route() {
+  while read -r iface dest gw flags refcnt use metric mask mtu window irtt; do
+    if [ "$dest" = "00000000" ] && [ "$mask" = "00000000" ]; then
+      logmsg "Default route found"
+      return 0
+    fi
+    logmsg "waiting for default route"
+    sleep 1
+  done < /proc/net/route
+
+  return 1
+}
+
+#Prereqs
+setup_prereqs () {
+	modprobe tun                                                        
+	modprobe vhost_net    
+	modprobe fuse         
+	modprobe iscsi_tcp
+	#Needed for iscsi tools
+	mkdir -p /run/lock
+	/usr/sbin/iscsid start
+	mount --make-rshared /
+	setup_cgroup                                                   
+
+	#Check network and default routes are up
+	wait_for_default_route
+	check_network_connection
+}
+
+#Make sure all prereqs are set
+setup_prereqs
+                                                              
+
+#Is this ZFS block device ?
+if [ -b /dev/zvol/persist/clustered-storage ]; then
+        mount /dev/zvol/persist/clustered-storage /var/lib  ## This is where we persist the cluster components (k3s containers)
+fi
+
+HOSTNAME=`/bin/hostname`
+#Forever loop every 15 secs
+while true;
+do
+if [ ! -f /var/lib/all_components_initialized ]; then
+	if [ ! -f /var/lib/k3s_initialized ]; then
+		#/var/lib is where all kubernetes components get installed.
+		logmsg "Installing K3S version $K3S_VERSION"
```

Comment:
Seems you are mixing tabs and spaces in this script. Prefer to use spaces - it will also resolve the reported Yetus issues.